src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -599,16 +643,27 @@ def compute_nhd_routing_v02(
                         usgs_df_sub = pd.DataFrame()
                         nudging_positions_list = []
 
+                    # classify reaches as either waterbodies (1) or rivers (2)
+                    subn_reach_list_with_type = []
+                    for reaches in subn_reach_list:
+                        if set(reaches) & wbodies_segs:
+                            reach_type = 1  # type 1 for waterbody/lake
+                        else:
+                            reach_type = 0  # type 0 for reach
+
+                        reach_and_type_tuple = (reaches, reach_type)
+
+                        subn_reach_list_with_type.append(reach_and_type_tuple)
```

Comment:
addresses point 3 in https://github.com/NOAA-OWP/t-route/issues/321#issuecomment-853141327